default/scripting/species/common/weapons.py

```diff
@@ -1,152 +1,54 @@
-            SetMaxDamage(partname="SR_WEAPON_3_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 3),
-            SetMaxDamage(partname="SR_WEAPON_4_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 5),
-        ],
-    ),
-    EffectsGroup(
-        scope=IsSource,
-        activation=Ship
-        & Armed
-        & (DesignHasPart(name="FT_HANGAR_2") | DesignHasPart(name="FT_HANGAR_3") | DesignHasPart(name="FT_HANGAR_4")),
-        effects=[
-            SetMaxSecondaryStat(partname="FT_HANGAR_2", value=Value + FIGHTER_DAMAGE_FACTOR * 1.5),
-            SetMaxSecondaryStat(partname="FT_HANGAR_3", value=Value + FIGHTER_DAMAGE_FACTOR * 1.5),
-            SetMaxSecondaryStat(partname="FT_HANGAR_4", value=Value + FIGHTER_DAMAGE_FACTOR * 1.5),
-        ],
-    ),
-    EffectsGroup(
-        scope=IsSource,
-        activation=Ship & Armed & DesignHasPart(name="SR_WEAPON_0_1"),
-        effects=SetMaxSecondaryStat(partname="SR_WEAPON_0_1", value=Value + 1),
-    ),
-]
-ULTIMATE_WEAPONS = [
-    EffectsGroup(
-        description="ULTIMATE_WEAPONS_DESC",
-        scope=IsSource,
-        activation=Ship
-        & Armed
-        & (
-            DesignHasPart(name="SR_WEAPON_1_1")
-            | DesignHasPart(name="SR_WEAPON_2_1")
-            | DesignHasPart(name="SR_WEAPON_3_1")
-            | DesignHasPart(name="SR_WEAPON_4_1")
+        EffectsGroup(
+            scope=IsSource,
+            activation=Ship & Armed & DesignHasPart(name="SR_WEAPON_0_1"),
+            effects=[
+                SetMaxSecondaryStat(partname="SR_WEAPON_0_1", value=Value + tier_0),
+            ],
         ),
-        effects=[
-            SetMaxDamage(partname="SR_WEAPON_1_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 3),
-            SetMaxDamage(partname="SR_WEAPON_2_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 6),
-            SetMaxDamage(partname="SR_WEAPON_3_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 9),
-            SetMaxDamage(partname="SR_WEAPON_4_1", value=Value + SHIP_WEAPON_DAMAGE_FACTOR * 15),
-        ],
-    ),
-    EffectsGroup(
-        scope=IsSource,
-        activation=Ship
-        & Armed
-        & (DesignHasPart(name="FT_HANGAR_2") | DesignHasPart(name="FT_HANGAR_3") | DesignHasPart(name="FT_HANGAR_4")),
-        effects=[
-            SetMaxSecondaryStat(partname="FT_HANGAR_2", value=Value + FIGHTER_DAMAGE_FACTOR * 4.5),
-            SetMaxSecondaryStat(partname="FT_HANGAR_3", value=Value + FIGHTER_DAMAGE_FACTOR * 4.5),
-            SetMaxSecondaryStat(partname="FT_HANGAR_4", value=Value + FIGHTER_DAMAGE_FACTOR * 4.5),
-        ],
-    ),
-    EffectsGroup(
-        scope=IsSource,
-        activation=Ship & Armed & DesignHasPart(name="SR_WEAPON_0_1"),
-        effects=SetMaxSecondaryStat(partname="SR_WEAPON_0_1", value=Value + 3),
-    ),
-]
+    ]
+
+
+BAD_WEAPONS = _weapon(tag="BAD", tier_0=-1, tier_1=-1, tier_2=-2, tier_3=-3, tier_4=-5, hangar=-1.0)
```

Comment:
as far as I can see/remember the cascading direct weapon damage steps (tier_*) are always equidistant (NB i was not aware, that death ray does two steps). so if you refactor, I think something akin to the following  signature would be better
```
BAD_WEAPONS = _weapon(tag="BAD", main_weapon_step=-1, fighter_damage_step=-1.0)
GOOD_WEAPONS = _weapon(tag="GOOD", main_weapon_step=1, fighter_damage_step=1.5)
GREAT_WEAPONS = _weapon(tag="GREAT", main_weapon_step=2, fighter_damage_step=3.0)
ULTIMATE_WEAPONS = _weapon(tag="ULTIMATE", main_weapon_step=3, fighter_damage_step=4.5)
```
(thinking more about this, actually it is the step size which gets set)

Reply:
I'll make it as a separate request, this one is already too large.